weave-js/src/components/Panel2/PanelPlot/plotState.ts

```diff
@@ -1478,3 +1505,60 @@ export function selectionContainsValue(
     return selection.includes(value);
   }
 }
+
+export function setThroughArray(
+  object: any,
+  path: Array<string | number>,
+  value: any,
+  broadcast: boolean = true
+): any {
+  if (path.length === 0) {
+    return object;
+  }
+
+  const [first, ...rest] = path;
+
+  if (first === '#') {
+    if (Array.isArray(object)) {
+      object.forEach((item: any, index: number) => {
+        setThroughArray(
+          item,
+          rest,
+          broadcast ? value : value[index],
+          broadcast
+        );
+      });
+    }
+  } else if (rest.length === 0) {
+    object[first] = value;
+  } else if (typeof object[first] === 'object' && object[first] !== null) {
+    setThroughArray(object[first], rest, value, broadcast);
+  }
+
+  return object;
+}
+
+export function getThroughArray(
+  object: any,
+  path: Array<string | number>
+): any {
+  if (path.length === 0) {
+    return object;
+  }
+
+  const [first, ...rest] = path;
+
+  if (first === '#') {
+    if (Array.isArray(object)) {
+      return object.map((item: any) => getThroughArray(item, rest));
+    } else {
+      return undefined;
+    }
+  } else if (typeof object[first] === 'object' && object[first] !== null) {
```

Comment:
I think this might error if object is an array but first is a number and that number is out of bounds.

Reply:
nit: i think this function could be written with less conditions if the base case test `path.length === 0` included a null check on object, then these non-# conditions might just reduce to always returning `getThroughArray(object[first], rest)`.

Reply:
done